scipy/sparse/extract.py

```diff
@@ -35,11 +35,11 @@ def find(A):
 
     """
 
-    A = coo_matrix(A).tocsr()  # sums duplicates
-    A.eliminate_zeros()        # removes explicit zeros
-    A = A.tocoo(copy=False)    # (cheaply) convert to COO
-
-    return A.row,A.col,A.data
+    A = coo_matrix(A, copy=False)
```

Comment:
True, though I believe the (unofficial) current policy for `scipy.sparse` is to canonicalize lazily and in-place, because the canonical form should be the common case. I'm happy to change this if it's likely to break anyone's code, though.

@pv, your thoughts?